docs/design/security/02-jet-job-security.md

```diff
@@ -167,6 +167,83 @@ obtain the PMSs and from them the required permissions. For each
 permission we ask the security context if the endpoint has the required
 permissions.
 
+### SecuredFunction
+
+There are some lambdas on the classpath which can be used to gain
+access to data-structures. For example:
+
+```java
+context -> context.hazelcastInstance().getMap(name);
+```
+
+We've introduced below interface which returns a list of permissions
+required to execute the function.
+```java
+@PrivateApi
+public interface SecuredFunction {
+
+    /**
+     * @return the list of permissions required to run this function
+     */
+    @Nullable
+    default List<Permission> permissions() {
+        return null;
+    }
+}
+```
+
+These permissions are checked at the initialization phase of the job,
+the permissions of the `ProcessorMetaSupplier` are checked at the
+submit phase of the job.
+
+You can find these functions in `SecuredFunctions` utility class.
+Any lambda we provide in the distribution which accesses a restricted
+data-structure should implement this interface.
```

Comment:
Should we point out that this must not return a value of a field that is serialized and sent to the server, but instead it should be hardcoded. Anyone implementing `SecuredFunction` reading this document will benefit from this info.